specification/07-theory_operations.adoc

```diff
@@ -876,6 +876,37 @@ TVM ->> TVM: Use device interface
 
 === Interface Unbinding
 
+=== Session Key Update
+
+In TEE-IO architecture, there are 2 sessions established between the host TSM
+and the device: SPDM session and IDE session. Both sessions use AES-GCM-256
+keys to protect the data. As such, the host software needs to update the
+session keys before the use of keys reach to the limit.
+
+==== SPDM Session Key Update
+
+SPDM session key update is managed by SPDM KEY_UPDATE command.
+The TSM shall count the usage of SPDM session keys, also knowns as the 64bit
+`sequence number` defined in <<SecuredSPDM>>.
+Whenever the VMM asks the TSM to encrypt the SPDM message, the TSM shall check
+if the `sequence number` reaches to the limit. If yes, the TSM shall notify VMM to
+perform SPDM Key Update action (through the `sbi_covh_spdm_key_update()` `COVH` call).
```

Comment:
The VMM never explictly asks the TSM to encrypt SPDM messages, but sends `COVH` requests that could lead to the generation and encryption of SPDM messages from the TSM. See https://github.com/riscv-non-isa/riscv-ap-tee-io/blob/main/specification/07-theory_operations.adoc#spdm-transport

So I suggest this instead:

```suggestion
Whenever the TSM generates and encrypts SPDM messages to support the host supervisor domain manager `COVH` requests, it shall check if the `sequence number` reaches to the limit. If it does, the TSM notifies the VMM to initiate an SPDM Key Update (through the `sbi_covh_spdm_key_update()` `COVH` call), by replying with an `SBI` error to the corresponding `COVH` request.
Once the SPDM Key Update completes, the VMM shall re-send the failing `COVH` request. 
```

Reply:
done